### ISSUE TITLE
Spring Cloud 2024.0.0 not compatible with SpringBoot 3.4.0

### ISSUE THREAD

**Issue by sabex** [spring-cloud/spring-cloud-release#300]
**Describe the bug**
SpringBoot 3.4.0 with Spring cloud 2024.0.0
Application Context fails at startup with message
`Caused by: org.springframework.cloud.configuration.CompatibilityNotMetException: Spring Cloud/ Spring Boot version compatibility checks have failed: [[VerificationResult@25e7f445 description = 'Spring Boot [3.4.0] is not compatible with this Spring Cloud release train', action = 'Change Spring Boot version to one of the following versions [3.2.x, 3.3.x] .`

Wiki: https://github.com/spring-cloud/spring-cloud-release/wiki/Supported-Versions#supported-releases
states these are compatible

```
<dependency>
                <groupId>org.springframework.cloud</groupId>
                <artifactId>spring-cloud-dependencies</artifactId>
                <version>${spring-cloud.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
```
where `<spring-cloud.version>2024.0.0</spring-cloud.version>`

```
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>3.4.0</version>
        <relativePath/> <!-- lookup parent from repository -->
    </parent>
```

**Comment by sabex** [comment]
I am unable to add files - I suspect this is due to a corporate security policy

**Comment by spencergibb** [comment]
My guess is you have mixed dependencies.

This project from start.spring.io starts fine

https://start.spring.io/#!type=maven-project&language=java&platformVersion=3.4.0&packaging=jar&jvmVersion=17&groupId=com.example&artifactId=democloud2024&name=democloud2024&description=Demo%20project%20for%20Spring%20Boot&packageName=com.example.democloud2024&dependencies=cloud-gateway

Check `mvn dependency:tree` and see what is there.

**Comment by sabex** [comment]
@spencergibb  thanks for checking, I did not think of the initralizr route and I agree its an issue elsewhere in my project.  nothing in the dependency tree jumped out at me, apart from some old AWS SDK jars which I am in the process of updated to v2

**Comment by sabex** [comment]
I think this may be my issue:
INFO] +- org.springframework.cloud:spring-cloud-starter-openfeign:jar:4.2.0:compile
[INFO] |  +- org.springframework.cloud:spring-cloud-starter:jar:4.1.0:compile
[INFO] |  |  +- org.springframework.cloud:spring-cloud-context:jar:4.1.0:compile
[INFO] |  |  \- org.springframework.security:spring-security-rsa:jar:1.1.1:compile
[INFO] |  |     \- org.bouncycastle:bcprov-jdk18on:jar:1.74:compile

**Comment by spencergibb** [comment]
Yup, something in your end is bringing in the older versions. The openfeign starter brings in the correct versions https://central.sonatype.com/artifact/org.springframework.cloud/spring-cloud-starter-openfeign/4.2.0/dependencies

**Comment by sabex** [comment]
its weird because I use that starter, and the 3 jars I have issues with are not mentioned anywhere else in the tree. 
I added explicit overrides to get past it.
